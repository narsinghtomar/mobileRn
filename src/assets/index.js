/**
 * Images
 */

import {default as IMAGES} from './images';

export {IMAGES};
