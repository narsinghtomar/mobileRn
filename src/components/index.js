/**
 * Import Components
 */
import {SnackBar, HideSnackBar} from './snackBar';
import Loader from './loader';
import Space from './space';
import {ButtonBlock, Button, IconButton} from './buttons';
import ImageWithText from './imageText';
import {InputField} from './inputFeild';
import HeaderBar from './headerBar';
import FloatingButton from './floatingButton';
/**
 * Export
 */
export {
  SnackBar,
  HideSnackBar,
  Loader,
  Space,
  ButtonBlock,
  Button,
  IconButton,
  ImageWithText,
  InputField,
  HeaderBar,
  FloatingButton,
};
