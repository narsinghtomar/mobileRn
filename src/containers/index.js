/**
 * Import Containers
 */
import OnBoardingPage from './onBoarding';
import LoginPage from './login';
import SignUpPage from './signUp';
import CategorizedListPage from './categorized';
import ArticleListPage from './articleList';
import WebViewPage from './webView';
import SearchListPage from './searchList';
import SearchArticlePage from './searchArticle';

/**
 * Export Containers
 */

export {
  OnBoardingPage,
  LoginPage,
  SignUpPage,
  CategorizedListPage,
  ArticleListPage,
  WebViewPage,
  SearchListPage,
  SearchArticlePage,
};
