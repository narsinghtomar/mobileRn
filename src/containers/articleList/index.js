/**
 * ArticleListPage Page
 */
import React, {useCallback, useEffect} from 'react';
import {View, FlatList} from 'react-native';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import * as homeActions from '../../redux/actions/articleList';
import {FloatingButton, HeaderBar, Loader} from '../../components';
import styles from './styles';
import ArticleItem from './articleItem';
import {APIKEY} from '../../utils/constants/constants';
import {SEARCH_LIST_PAGE_ROUTE} from '../../navigation/StackNavigator';

function ArticleListPage(props) {
  const {
    navigation,
    fetchNytimesData,
    isFatching,
    nyTimesInfo = [],
    route,
  } = props || {};
  const {params} = route || {};
  const {category} = params || {};

  console.log('nyTimesInfo', nyTimesInfo?.length);

  useEffect(() => {
    getNyTimesData();
  }, [category, getNyTimesData]);

  const getNyTimesData = useCallback(() => {
    const categoryData = {
      category: category,
      apiKey: APIKEY,
    };
    fetchNytimesData && fetchNytimesData({categoryData});
  }, [fetchNytimesData, category]);

  /**
   * Render Inspiring Quote Item
   */
  const renderArticleItem = ({item, index}) => {
    return <ArticleItem article={item} index={index} navigation={navigation} />;
  };

  const onFlotingBtnPress = () => {
    navigation.navigate(SEARCH_LIST_PAGE_ROUTE, {
      category: category,
    });
  };

  return (
    <View style={styles.container}>
      <Loader isLoading={isFatching} />
      <HeaderBar headerText={category || ''} navigation={navigation} />
      <View style={styles.flatListContainer}>
        <FlatList
          showsVerticalScrollIndicator={false}
          data={nyTimesInfo}
          keyExtractor={(_item, index) => index.toString()}
          renderItem={renderArticleItem}
          extraData={nyTimesInfo}
          style={styles.flatListBox}
        />
      </View>
      <FloatingButton onPress={onFlotingBtnPress} />
    </View>
  );
}

/**
 *
 * @param {*} dispatch
 * @returns
 */
const mapDispatchToProps = dispatch => {
  return bindActionCreators(Object.assign({}, homeActions), dispatch);
};

/**
 *
 * @param {*} state
 * @returns
 */
const mapStateToProps = state => ({
  isFatching: state.articleListReducer.isFatching,
  nyTimesInfo: state.articleListReducer.nyTimesInfo,
  getNyTimesErrorData: state.articleListReducer.getNyTimesErrorData,
});
export default connect(mapStateToProps, mapDispatchToProps)(ArticleListPage);
