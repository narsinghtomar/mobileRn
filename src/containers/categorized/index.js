/**
 * CategorizedList Page
 */
import React from 'react';
import {View, Text, SafeAreaView} from 'react-native';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';

import {
  ARTICLE_LIST_PAGE_ROUTE,
  SEARCH_LIST_PAGE_ROUTE,
} from '../../navigation/StackNavigator';
import {SCIENCE, WORLD} from '../../utils/constants/constants';
import {ButtonBlock, FloatingButton, Space} from '../../components';
import {strings} from '../../i18n';
import styles from './styles';

function CategorizedListPage(props) {
  const {navigation} = props || {};

  /**
   * World Button Props
   */
  const worldButtonProps = category => {
    return {
      text: strings('category.world'),
      onPress: () => {
        getStartedPress(category);
      },
      buttonStyle: styles.btnContainer,
    };
  };

  /**
   * Science Button Props
   */
  const scienceButtonProps = category => {
    return {
      text: strings('category.science'),
      onPress: () => {
        getStartedPress(category);
      },
      buttonStyle: styles.btnContainer,
    };
  };

  const getStartedPress = category => {
    navigation.navigate(ARTICLE_LIST_PAGE_ROUTE, {
      category: category,
    });
  };

  const onFlotingBtnPress = () => {
    navigation.navigate(SEARCH_LIST_PAGE_ROUTE);
  };

  return (
    <SafeAreaView style={styles.container}>
      <View style={styles.baseContainer}>
        <Text style={styles.baseTitle}>
          {strings('category.selectCategoryMsg')}
        </Text>
      </View>
      <Space height={45} />
      <View>{ButtonBlock(scienceButtonProps(SCIENCE))}</View>
      <View>{ButtonBlock(worldButtonProps(WORLD))}</View>
      <Space height={50} />
      <FloatingButton onPress={onFlotingBtnPress} />
    </SafeAreaView>
  );
}

/**
 *
 * @param {*} dispatch
 * @returns
 */
const mapDispatchToProps = dispatch => {
  return bindActionCreators(Object.assign({}), dispatch);
};

/**
 *
 * @param {*} _state
 * @returns
 */
const mapStateToProps = _state => ({});
export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(CategorizedListPage);
