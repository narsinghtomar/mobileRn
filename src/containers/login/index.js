/**
 * Login Page
 */
import React, {useEffect, useState} from 'react';
import {
  View,
  Text,
  TouchableWithoutFeedback,
  KeyboardAvoidingView,
  Keyboard,
  TouchableOpacity,
} from 'react-native';
import {bindActionCreators} from 'redux';
import {connect} from 'react-redux';

import {ButtonBlock, InputField, Loader, Space} from '../../components';
import {strings} from '../../i18n';
import {useFormik} from 'formik';
import {
  CATEGORIZEDLIST_PAGE_ROUTE,
  SIGNUP_PAGE_ROUTE,
} from '../../navigation/StackNavigator';
import {generateHeaders} from '../../utils/constants/apiHeaders';
import {loginValidationSchema} from '../../utils/vallidator';
import * as loginActions from '../../redux/actions/login';
import {isIOS} from '../../utils/helper';
import {showAlert} from '../../utils';
import styles from './styles';

/**
 *
 * @param {*} props
 * @returns
 */
function LoginPage(props) {
  const {navigation, loginUser, isFatching, loginInfo, loginErrorData} =
    props || {};

  const [isSignInCalled, setIsSignInCalled] = useState(false);

  /**
   * UseFormik for the Validation
   */
  const {handleChange, handleSubmit, setFieldTouched, values, errors, touched} =
    useFormik({
      validationSchema: loginValidationSchema,
      initialValues: {email: '', password: ''},
      onSubmit: userValues => {
        signIn(userValues?.email ?? '', userValues?.password ?? '');
      },
    });

  /**
   *  SignInUser : API Calls by Action
   * @param {*} email
   * @param {*} password
   */
  const signIn = (email, password) => {
    const loginData = {
      username: email,
      password: password,
    };
    const headers = generateHeaders();
    loginUser && loginUser({headers, loginData});
    navigation.navigate(CATEGORIZEDLIST_PAGE_ROUTE);
    setIsSignInCalled(true);
  };

  /**
   * API Response handler
   * API SUCEESS & ERROR
   */
  useEffect(() => {
    if (isSignInCalled && loginInfo) {
      navigation.navigate(CATEGORIZEDLIST_PAGE_ROUTE);
    } else if (isSignInCalled && loginErrorData) {
      showAlert(strings('app.error'), strings('app.error'));
    }
  }, [loginErrorData, loginInfo, isSignInCalled, navigation]);

  /**
   * signInButtonProps
   */
  const signInButtonProps = () => {
    return {
      text: strings('login.singIn'),
      onPress: handleSubmit,
    };
  };

  /**
   * emailProps
   */
  const emailProps = () => {
    return {
      labelText: strings('login.email'),
      textInputName: 'email',
      keyboardType: 'email-address',
      onChangeText: text => {
        handleChange('email')(text);
      },
      onBlur: () => setFieldTouched('email'),
      value: values.email,
      errorMessage: errors.email,
      isErrorMsgRequired: touched.email && errors.email,
    };
  };

  /**
   * passwordProps
   */
  const passwordProps = () => {
    return {
      labelText: strings('login.password'),
      textInputName: 'password',
      onChangeText: text => {
        handleChange('password')(text);
      },
      onBlur: () => setFieldTouched('password'),
      secureTextEntry: true,
      value: values.password,
      errorMessage: errors.password,
      isErrorMsgRequired: touched.password && errors.password,
    };
  };

  /**
   * creatAccountView
   */
  const creatAccountView = () => {
    return (
      <View style={styles.signUpView}>
        <Text style={styles.noAccountText}>{strings('login.noAccount')}</Text>
        <TouchableOpacity activeOpacity={0.5} onPress={onSignUpPress}>
          <Text style={styles.noAccountTextTap}>
            {strings('login.noAccountSignUp')}
          </Text>
        </TouchableOpacity>
      </View>
    );
  };

  /**
   * onSignUpPress
   */
  const onSignUpPress = () => {
    navigation.navigate(SIGNUP_PAGE_ROUTE);
  };

  /**
   *  Return
   */
  return (
    <View style={styles.mainContainer}>
      <Loader isLoading={isFatching} />
      <TouchableWithoutFeedback onPress={Keyboard.dismiss} accessible={false}>
        <KeyboardAvoidingView
          style={styles.avoidingView}
          behavior={isIOS() ? 'padding' : null}
          keyboardVerticalOffset={10}>
          <View style={styles.container}>
            <Text style={styles.signInText}>{strings('login.singIn')}</Text>
            <View>
              {InputField(emailProps())}
              <Space height={10} />
              {InputField(passwordProps())}
            </View>
            <View>
              {ButtonBlock(signInButtonProps())}
              {creatAccountView()}
            </View>
          </View>
        </KeyboardAvoidingView>
      </TouchableWithoutFeedback>
    </View>
  );
}

/**
 *
 * @param {*} dispatch
 * @returns
 */
const mapDispatchToProps = dispatch => {
  return bindActionCreators(Object.assign({}, loginActions), dispatch);
};

/**
 *
 * @param {*} state
 * @returns
 */
const mapStateToProps = state => ({
  isFatching: state.loginReducer.isFatching,
  loginInfo: state.loginReducer.loginInfo,
  loginErrorData: state.loginReducer.loginErrorData,
});
export default connect(mapStateToProps, mapDispatchToProps)(LoginPage);
