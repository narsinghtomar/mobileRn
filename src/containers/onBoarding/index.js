/**
 * OnBoarding Page
 */
import React from 'react';
import {View, Text, Image, SafeAreaView} from 'react-native';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';

import {LOGIN_PAGE_ROUTE} from '../../navigation/StackNavigator';
import {ButtonBlock, Space} from '../../components';
import {IMAGES} from '../../assets';
import {strings} from '../../i18n';
import styles from './styles';

function OnBoardingPage(props) {
  const {navigation} = props || {};

  /**
   * Get Started Button Props
   */
  const getStartedButtonProps = () => {
    return {
      text: strings('onBoarding.getStarted'),
      onPress: getStartedPress,
      buttonStyle: styles.btnContainer,
    };
  };
  const getStartedPress = () => {
    navigation.navigate(LOGIN_PAGE_ROUTE);
  };

  /**
   * _buildHeaderView
   */
  const _buildHeaderView = () => {
    return (
      <View style={styles.headerContainer}>
        <Image
          source={IMAGES.TEMPLOGO}
          style={styles.imgStyle}
          resizeMode="contain"
        />
      </View>
    );
  };

  return (
    <SafeAreaView style={styles.container}>
      <Space height={60} />
      {_buildHeaderView()}
      <Space height={60} />
      <View style={styles.baseContainer}>
        <Text style={styles.baseTitle}>
          {strings('onBoarding.bulletTitle')}
        </Text>
      </View>
      <Space height={45} />
      <View>{ButtonBlock(getStartedButtonProps())}</View>
    </SafeAreaView>
  );
}

/**
 *
 * @param {*} dispatch
 * @returns
 */
const mapDispatchToProps = dispatch => {
  return bindActionCreators(Object.assign({}), dispatch);
};

/**
 *
 * @param {*} _state
 * @returns
 */
const mapStateToProps = _state => ({});
export default connect(mapStateToProps, mapDispatchToProps)(OnBoardingPage);
