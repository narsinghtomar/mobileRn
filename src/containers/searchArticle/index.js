/**
 * SearchArticles Page
 */
import React, {useState, useCallback, useEffect} from 'react';
import {View, FlatList, ActivityIndicator, Text} from 'react-native';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import * as searchArticlesAction from '../../redux/actions/searchArticle';
import {HeaderBar, Loader, Space} from '../../components';
import ArticleItem from '../articleList/articleItem';
import {APIKEY, SEARCH, WORLD} from '../../utils/constants/constants';
import styles from './styles';
import {COLOR_SECONDARY} from '../../utils/colors';
import {strings} from '../../i18n';

function SearchArticlePage(props) {
  const {
    navigation,
    searchArticles,
    isFatching,
    searchArticlesData = [],
    route,
    isPaginating,
    isPageFinished,
    articlesCount,
  } = props || {};
  const {params} = route || {};
  const {category, searchText} = params || {};

  const [page, setPage] = useState(0);
  console.log('searchArticlesData', searchArticlesData?.length);

  /**
   * API CALLs
   * Page With default size Zero for Articles list
   */
  useEffect(() => {
    setPage(0);
    const delay = setTimeout(() => {
      fetchSearchArticles(0);
    }, 50);
    return () => {
      clearTimeout(delay);
    };
  }, [category, fetchSearchArticles]);

  /**
   * method for handling pagination when user reach to end of the list
   */
  const handleLoadMoreData = useCallback(() => {
    if (!isPaginating && !isPageFinished) {
      const nextPage = page + 1;
      setPage(nextPage);
      fetchSearchArticles(nextPage);
    }
  }, [isPaginating, isPageFinished, setPage, page, fetchSearchArticles]);

  const fetchSearchArticles = useCallback(
    (pageNumber = 0) => {
      const searchData = {
        category: category ? category : WORLD,
        apiKey: APIKEY,
        searchQuary: searchText,
        page: pageNumber,
        pageSize: 10,
      };
      searchArticles && searchArticles({searchData});
    },
    [searchArticles, category, searchText],
  );

  /**
   * Returns footer section with ActivityIndicator
   */
  const listFooter = () => {
    if (isPaginating && !isPageFinished) {
      return <ActivityIndicator size={'large'} color={COLOR_SECONDARY} />;
    }
    if (articlesCount === 1) {
      return <Space height={100} />;
    } else {
      return <Space height={70} />;
    }
  };

  /**
   * Render Inspiring Quote Item
   */
  const renderArticleItem = ({item, index}) => {
    return (
      <ArticleItem
        article={item}
        index={index}
        navigation={navigation}
        type={SEARCH}
      />
    );
  };

  /**
   * _buildRenderEmptyMsg
   */
  const _buildRenderEmptyMsg = () => {
    return searchArticlesData?.length === 0 && !isFatching ? (
      <View style={styles.notFoundContainer}>
        <Text style={styles.notFoundText}>
          {strings('category.resultNotFound')}
        </Text>
      </View>
    ) : null;
  };

  return (
    <View style={styles.container}>
      <Loader isLoading={isFatching} />
      <HeaderBar headerText={category || ''} navigation={navigation} />
      {_buildRenderEmptyMsg()}
      <View style={styles.flatListContainer}>
        <FlatList
          showsVerticalScrollIndicator={false}
          data={searchArticlesData}
          keyExtractor={(_item, index) => index.toString()}
          renderItem={renderArticleItem}
          extraData={searchArticlesData}
          style={styles.flatListBox}
          onEndReached={handleLoadMoreData}
          onEndReachedThreshold={0.1}
          ListFooterComponent={listFooter}
        />
      </View>
    </View>
  );
}

/**
 *
 * @param {*} dispatch
 * @returns
 */
const mapDispatchToProps = dispatch => {
  return bindActionCreators(Object.assign({}, searchArticlesAction), dispatch);
};

/**
 *
 * @param {*} state
 * @returns
 */
const mapStateToProps = state => ({
  isFatching: state.searchArticlesReducer.isFatching,
  searchArticlesData: state.searchArticlesReducer.searchArticlesData,
  searchArticlesErrorData: state.searchArticlesReducer.searchArticlesErrorData,
  isPaginating: state.searchArticlesReducer.isPaginating,
  isPageFinished: state.searchArticlesReducer.isPageFinished,
  articlesCount: state.searchArticlesReducer.articlesCount,
});
export default connect(mapStateToProps, mapDispatchToProps)(SearchArticlePage);
