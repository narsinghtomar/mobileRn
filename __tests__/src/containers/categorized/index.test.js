/**
 * Button
 */
import 'react-native';
import React from 'react';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import {CategorizedListPage} from '../../../../src/containers';
import renderer from 'react-test-renderer';

describe('CategorizedList Page', () => {
  const initialState = {};

  const mockStore = configureStore();
  let store;

  it('CategorizedList page renders correctly', async () => {
    jest.useFakeTimers();
    store = mockStore(initialState);
    const tree = await renderer
      .create(
        <Provider store={store}>
          <CategorizedListPage />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
