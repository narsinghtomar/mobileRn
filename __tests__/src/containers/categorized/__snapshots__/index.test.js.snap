// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategorizedList Page CategorizedList page renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "marginHorizontal": 41,
        "textAlign": "left",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#6F10A2",
          "fontSize": 16,
          "fontWeight": "700",
          "lineHeight": 24,
          "paddingHorizontal": 20,
          "textAlign": "center",
        }
      }
    >
      [missing "en.category.selectCategoryMsg" translation]
    </Text>
  </View>
  <View
    style={
      Object {
        "height": 45,
        "width": undefined,
      }
    }
  />
  <View>
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#6F10A2",
          "borderRadius": 4,
          "height": 48,
          "justifyContent": "center",
          "marginTop": 10,
          "opacity": 1,
          "paddingHorizontal": 12,
          "width": 412.50000000000006,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "alignSelf": "center",
              "color": "#FFFFFF",
              "fontSize": 14,
              "fontWeight": "700",
              "letterSpacing": 0.25,
              "textAlign": "center",
              "textAlignVertical": "center",
            },
            undefined,
          ]
        }
      >
        [missing "en.category.science" translation]
      </Text>
    </View>
  </View>
  <View>
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#6F10A2",
          "borderRadius": 4,
          "height": 48,
          "justifyContent": "center",
          "marginTop": 10,
          "opacity": 1,
          "paddingHorizontal": 12,
          "width": 412.50000000000006,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "alignSelf": "center",
              "color": "#FFFFFF",
              "fontSize": 14,
              "fontWeight": "700",
              "letterSpacing": 0.25,
              "textAlign": "center",
              "textAlignVertical": "center",
            },
            undefined,
          ]
        }
      >
        [missing "en.category.world" translation]
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "height": 50,
        "width": undefined,
      }
    }
  />
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#6F10A2",
        "borderColor": "#C4C4C4",
        "borderRadius": 40,
        "borderWidth": 0.4,
        "bottom": 30,
        "height": 60,
        "justifyContent": "center",
        "opacity": 1,
        "position": "absolute",
        "right": 30,
        "width": 60,
      }
    }
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../src/assets/images/search/search.png",
        }
      }
      style={
        Object {
          "height": 30,
          "width": 30,
        }
      }
    />
  </View>
</RCTSafeAreaView>
`;
